pkg/pillar/devicenetwork/dnc.go

```diff
@@ -429,7 +434,10 @@ func VerifyDevicePortConfig(ctx *DeviceNetworkContext) {
 					pending.PendDPC.Key)
 			}
 			passed = true
-			if ctx.NextDPCIndex == 0 {
+
+			// If there are port level errors in current selected DPC, we should mark it
+			// for re-test during the next TestBetterTimer innvocation.
+			if ctx.NextDPCIndex == 0 && !pending.PendDNS.HasErrors() {
```

Comment:
We don't need TestBetterTImer - the 5 minute test timer will reverify and update the status.

Reply:
5 minute test timer clears the errors when the port is mgmt and free. Mgmt and non-free port errors may also get cleared whenn none of the mgmt-free ports succeed the test. But, when there are errors in mgmt-nonfree ports and app-shared ports (dhcp may not have started in the earlier run due to errors), we will have to re-test the entire DPC.